### DISCUSSION TITLE
The detailed explanation of externalizeDepsPlugin

### DISCUSSION THREAD

**Discussion by Oleksii14** [alex8088/electron-vite#146] [General]
Hey! I would like to ask you to explain why do we need to "externalize" dependencies from main/preload/renderer. I read two sections in the docs ([1](https://evite.netlify.app/guide/dev.html#dependencies-vs-devdependencies) , [2](https://evite.netlify.app/guide/build.html#externals)) and it is still not fully clear, for example, why do we need to put real app's dependencies to `devDependencies`.

I would appreciate if you provide more detailed examples or cases

**Comment by alex8088** [comment]
We cannot avoid using third-party modules during development. These modules usually support CJS, so we don't need to rebuild and bundle them, we only need to deal with our own code. This is called "externalization". Of course, we can also bundled them, but you can imagine how big the final chunk will be, which is extremely unfavorable for module loading. 

However, these modules are required by the app, so they must be installed in `dependencies`, so that they will be packaged together by the packaging tool, and will not be lost. But if installed in `devDependencies`, packaging tools are usually not packaged.

**Reply by Oleksii14** [reply]
>However, these modules are required by the app, so they must be installed in dependencies, so that they will be packaged together by the packaging tool, and will not be lost.

In this case, why is it recommended to install dependencies to `devDepencies` in the [docs](https://evite.netlify.app/guide/build.html#externals)?

<img width="715" alt="image" src="https://user-images.githubusercontent.com/36737084/232430526-f73ff76c-d344-4259-b313-269fdf10037d.png">

**Reply by alex8088** [reply]
You should know whether your module is used by the main process or the renderer process. The renderer development is the same as regular web app development  and requires full bundling. And the build standard is iife. After these modules are bundled, they are no longer needed because they have been bundled into  the renderer chunks. So install them in `devDep`, then the packaging tool will not pack them to app.

In addition, there is no need to use `externalizeDepPlugin`  in the renderer vite config.
